### PR TITLE
Ap 840 client assets

### PR DESCRIPTION
**Pull request by willc-work**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-840)

Added attributes to check application and updates the XML accordingly.

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/services/ccms/attribute_value_generator.rb [hunk 1]
```diff
@@ -92,6 +87,42 @@ def app_amendment_type(_options)
       @legal_aid_application.used_delegated_functions? ? 'SUBDP' : 'SUB'
     end
 
+    def client_owns_vehicle(_options)
+      @legal_aid_application.own_vehicle? ? 'true' : 'false'
+    end
```
Comment (reggieb) [inline app/services/ccms/attribute_value_generator.rb]: Do `true` and `false` have to be returned as strings? If they can be returned as Booleans, this whole section of code gets a lot easier. For example this just becomes:
```ruby
def client_owns_vehicle(_options)
  @legal_aid_application.own_vehicle?
end
```

File: app/services/ccms/attribute_value_generator.rb [hunk 2]
```diff
@@ -92,6 +87,42 @@ def app_amendment_type(_options)
...
+
+    def client_owed_money(_options)
+      @legal_aid_application.other_assets_declaration.money_owed_value.nil? ? 'false' : 'true'
```
Comment (reggieb) [inline app/services/ccms/attribute_value_generator.rb]: There is a lot of repetition here of the pattern `x.nil? ? 'true' : 'false'`. Firstly, if we can return Booleans, could this simply be:
```ruby
@legal_aid_application.other_assets_declaration.money_owed_value.present?
```
If the pattern used is really the only way to do it, I'd suggest wrapping in a method:
```ruby
def not_nil?(value)
  value.nil? ? 'false' : 'true'
end

def client_owed_money(_options)
  not_nil? @legal_aid_application.other_assets_declaration.money_owed_value
end
```
etc.

File: config/ccms/ccms_keys.yml [hunk 3]
```diff
@@ -409,7 +409,7 @@ global_means:
     :response_type: text
     :user_defined: false
   GB_INPUT_B_16WP2_7A:
-    :value: false
+    :value: '#client_has_interest_in_a_trust'
     :br100_meaning: 'Trust: The client has an interest in a trust?'
     :response_type: boolean
```
Comment (reggieb) [inline config/ccms/ccms_keys.yml]: Boolean!

File: app/services/ccms/attribute_value_generator.rb (hunk 2 above)
Comment (reggieb) [reply app/services/ccms/attribute_value_generator.rb]: I've just had a bit of a play:
```ruby
x = Nokogiri::XML('<body><foo></foo><bar></bar></body>')

f = x.at_xpath '/body/foo'
f.content = 'true'
f.set_attribute 'bol', 'true'

b = x.at_xpath '/body/bar'
b.content = true
b.set_attribute 'bol', true

puts x.to_s
```
Outputs:
```xml
<?xml version="1.0"?>
<body>
  <foo bol="true">true</foo>
  <bar bol="true">true</bar>
</body>
```
As you can see setting `true` or `"true"` results in the same XML output.

Comment (reggieb) [reply app/services/ccms/attribute_value_generator.rb]: So remove all the text "true" and "false" and just use boolean operations `.nil?`, `.blank?` and `.present?` etc.

Comment (willc-work) [reply app/services/ccms/attribute_value_generator.rb]: yes all works, thank you much cleaner

**Pull request merged**